core/common/src/geometry/CoordinateReferenceSystem.ts

```diff
@@ -0,0 +1,457 @@
+      this.unit = _data.unit;
+      this.projection = _data.projection ? Projection.fromJSON(_data.projection) : undefined;
+      this.area = _data.area ? HorizontalCRSArea.fromJSON(_data.area) : undefined;
+    }
+  }
+
+  /** @internal */
+  public static fromJSON(data: HorizontalCRSProps): HorizontalCRS {
+    return new HorizontalCRS(data);
+  }
+
+  /** @internal */
+  public toJSON(): HorizontalCRSProps {
+    const data: HorizontalCRSProps = {};
+    data.id = this.id;
+    data.description = this.description;
+    data.source = this.source;
+    data.deprecated = this.deprecated;
+    data.epsg = this.epsg;
+    data.datumId = this.datumId;
+    data.datum = this.datum ? this.datum.toJSON() : undefined;
+    data.ellipsoidId = this.ellipsoidId;
+    data.ellipsoid = this.ellipsoid ? this.ellipsoid.toJSON() : undefined;
+    data.unit = this.unit;
+    data.projection = this.projection ? this.projection.toJSON() : undefined;
+    data.area = this.area ? this.area.toJSON() : undefined;
+    return data;
+  }
+
+  /** @internal */
+  public equals(other: HorizontalCRS): boolean {
+    if (this.id !== other.id ||
+      this.description !== other.description ||
+      this.source !== other.source ||
+      this.deprecated !== other.deprecated ||
+      this.epsg !== other.epsg ||
+      this.datumId !== other.datumId ||
+      this.ellipsoidId !== other.ellipsoidId ||
+      this.unit !== other.unit)
+      return false;
+
+    if ((this.datum === undefined) !== (other.datum === undefined))
+      return false;
+
+    if (this.datum && !this.datum.equals(other.datum!))
+      return false;
+
+    if ((this.ellipsoid === undefined) !== (other.ellipsoid === undefined))
+      return false;
+
+    if (this.ellipsoid && !this.ellipsoid.equals(other.ellipsoid!))
+      return false;
+
+    if ((this.projection === undefined) !== (other.projection === undefined))
+      return false;
+
+    if (this.projection && !this.projection.equals(other.projection!))
+      return false;
+
+    if ((this.area === undefined) !== (other.area === undefined))
+      return false;
+
+    if (this.area && !this.area.equals(other.area!))
+      return false;
+
+    return true;
+  }
+}
+
+/** Vertical Geographic Coordinate reference System definition
+ * @alpha
+ */
+export interface VerticalCRSProps {
+  /** Vertical CRS Key name */
+  id?: string;
+}
+
+/** Vertical Coordinate reference System implementation.
+ *  The VerticalCRS contains currently a single identifier property of string type. Although
+ *  we currently only support four distinct key values "GEOID", "ELLIPSOID", "NAVD88" and "NGVD29"
+ *  we expect to support a broader set in the future including, eventually, user defined vertical CRS
+ *  which will require additional parameters to be added.
+ *  @alpha
+*/
+export class VerticalCRS implements VerticalCRSProps {
+  /** Vertical CRS Key name. The only supported values are currently "GEOID", "ELLIPSOID", "NAVD88" and "NGVD29" */
```

Comment:
Consider using a union type `"GEOID" | "ELLIPSOID" | ...`. Can expand in future.

Reply:
All enums have been modified to string unions.